sources/api/schnauzer/src/helpers.rs

```diff
@@ -1346,9 +1361,188 @@ handlebars_helper!(any_enabled: |arg: Value| {
+        OciSpecSection::ResourceLimits => oci_spec_resource_limits(oci_defaults_values)?,
+    };
+
+    // Write out the final values to the configuration file
+    out.write(result_lines.as_str())
+        .context(error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
+}
+
+/// This helper writes out the capabilities section of the default
+/// OCI runtime spec that `containerd` will use.
+///
+/// This function is called by the `oci_defaults` helper function,
+/// specifically when matching over the `oci_spec_section` parameter
+/// to determine which section of the OCI spec to render.
+///
+/// This helper function generates the linux capabilities section of
+/// the OCI runtime spec from the provided `value` parameter, which is
+/// the settings data from the datastore (`settings.oci-defaults.capabilities`).
+fn oci_spec_capabilities(value: &Value) -> Result<String, RenderError> {
+    let oci_default_capabilities: HashMap<OciDefaultsCapability, bool> =
+        serde_json::from_value(value.clone())?;
+
+    // Only output the capabilities we support
+    let capabilities_lines: Vec<String> = oci_default_capabilities
+        .iter()
+        .filter(|(_, &capability_enabled)| capability_enabled)
+        .map(|(&capability, _)| format!("\"{}\"", capability.to_linux_string()))
+        .collect();
+
+    let capabilities_lines_joined = capabilities_lines.join(",\n");
+
+    let capabilities_section = format!(
+        "\"bounding\": [
+{capabilities_bounding}
+],
+\"effective\": [
+{capabilities_effective}
+],
+\"permitted\": [
+{capabilities_permitted}
+]",
+        capabilities_bounding = capabilities_lines_joined,
+        capabilities_effective = capabilities_lines_joined,
+        capabilities_permitted = capabilities_lines_joined,
+    );
+
+    debug!("capabilities_section: \n{}", capabilities_section);
+
+    Ok(capabilities_section)
+}
+
+/// This helper writes out the resource limits section of the default
+/// OCI runtime spec that `containerd` will use.
+///
+/// This function is called by the `oci_defaults` helper function,
+/// specifically when matching over the `oci_spec_section` parameter
+/// to determine which section of the OCI spec to render.
+///
+/// This helper function generates the resource limits section of
+/// the OCI runtime spec from the provided `value` parameter, which is
+/// the settings data from the datastore (`settings.oci-defaults.resource-limits`).
+fn oci_spec_resource_limits(value: &Value) -> Result<String, RenderError> {
+    let oci_default_rlimits: HashMap<OciDefaultsResourceLimitType, OciDefaultsResourceLimit> =
+        serde_json::from_value(value.clone())?;
+
+    let mut result_lines = String::new();
+    let mut is_first = true;
+    for (rlimit_type, values) in oci_default_rlimits
+        .iter()
+        // Bottlerocket model values are optional, if we have no hard_limit or soft_limit, we skip
+        // the map entry.
```

Comment:
We can make those fields not optional, is this something we'd like to error on during settings validation?

Reply:
We fixed this with `#[model(add_option = false)]` which eliminates these unsightly issues.